src/AutoMapper.php

```diff
@@ -53,28 +59,21 @@ class AutoMapper implements AutoMapperInterface, AutoMapperRegistryInterface, Ma
     public const RELEASE_VERSION = 0;
     public const EXTRA_VERSION = 'DEV';
 
-    /** @var MapperGeneratorMetadataInterface[] */
-    private array $metadata = [];
-
     /** @var GeneratedMapper[] */
     private array $mapperRegistry = [];
 
     public function __construct(
         private readonly ClassLoaderInterface $classLoader,
         private readonly ChainTransformerFactory $chainTransformerFactory,
-        public readonly CustomTransformersRegistry $customTransformersRegistry,
-        private readonly ?MapperGeneratorMetadataFactoryInterface $mapperConfigurationFactory = null
```

Comment:
do you know why the `MapperGeneratorMetadataFactoryInterface` dependency used to be optional? I can't see any case where it is justified